### PR TITLE
Create LND invoice

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

File: ios/BlueWallet/Info.plist [hunk 1]
```diff
@@ -56,7 +56,7 @@
 	<key>NSCalendarsUsageDescription</key>
 	<string>This alert should not show up as we do not require this data</string>
 	<key>NSCameraUsageDescription</key>
-	<string>In order to quickly scan the recipient&apos;s address, we need your permission to use the camera to scan their QR Code.</string>
+	<string>In order to quickly scan the recipient's address, we need your permission to use the camera to scan their QR Code.</string>
```
Comment (Overtorment) [inline ios/BlueWallet/Info.plist]: I see this changing back and forth several times

File: screen/lnd/lndCreateInvoice.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
+import React, { Component } from 'react';
+import { ActivityIndicator, View, TextInput, KeyboardAvoidingView, Keyboard, TouchableWithoutFeedback, Text } from 'react-native';
+import { BlueNavigationStyle, BlueButton } from '../../BlueComponents';
+import PropTypes from 'prop-types';
+import { BitcoinUnit } from '../../models/bitcoinUnits';
+import ReactNativeHapticFeedback from 'react-native-haptic-feedback';
+/** @type {AppStorage} */
```
Comment (Overtorment) [inline screen/lnd/lndCreateInvoice.js]: wrong typehint

File: screen/lnd/lndViewInvoice.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+import { Animated, StyleSheet, View, TouchableOpacity, Clipboard, Share } from 'react-native';
+// import { QRCode } from 'react-native-custom-qr-codes';
```
Comment (Overtorment) [inline screen/lnd/lndViewInvoice.js]: dead code

File: screen/lnd/lndViewInvoice.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    if (this.state.isLoading) {
+      return <BlueLoading />;
+    }
+
+    return (
+      <SafeBlueArea style={{ flex: 1 }}>
+        <View style={{ flex: 1, justifyContent: 'space-between', alignItems: 'center' }}>
+          <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', paddingHorizontal: 16 }}>
+            {/* <QRCode
+              content={this.state.invoice}
```
Comment (Overtorment) [inline screen/lnd/lndViewInvoice.js]: dead code

File: screen/lnd/scanLndInvoice.js [hunk 5]
```diff
@@ -72,7 +73,7 @@ export default class ScanLndInvoice extends React.Component {
 
     if (!this.state.fromWallet) {
       alert('Error: cant find source wallet (this should never happen)');
-      return this.props.navigation.goBack();
+      return this.props.navigation.dismiss();
```
Comment (Overtorment) [inline screen/lnd/scanLndInvoice.js]: is it better than `goBack` ..?

File: screen/lnd/scanLndInvoice.js [hunk 6]
```diff
@@ -275,7 +271,7 @@ export default class ScanLndInvoice extends React.Component {
 
 ScanLndInvoice.propTypes = {
   navigation: PropTypes.shape({
-    goBack: PropTypes.function,
+    dismiss: PropTypes.function,
```
Comment (Overtorment) [inline screen/lnd/scanLndInvoice.js]: I always wondered. What is this thing in the end of every screen? how does it help..?

File: screen/lnd/lndCreateInvoice.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+                  height: 44,
+                  marginHorizontal: 20,
+                  alignItems: 'center',
+                  marginVertical: 8,
+                  borderRadius: 4,
+                }}
+              >
+                <TextInput
+                  onChangeText={text => this.setState({ memo: text })}
+                  placeholder={loc.send.details.note_placeholder}
```
Comment (Overtorment) [inline screen/lnd/lndCreateInvoice.js]: should be some other text.  this is not 'note to self', its publicly visible to payer

File: screen/lnd/scanLndInvoice.js (hunk 6 above)
Comment (pajasevi) [reply screen/lnd/scanLndInvoice.js]: @Overtorment You mean Proptypes? It's React's way to check types in runtime. https://reactjs.org/docs/typechecking-with-proptypes.html

File: screen/lnd/scanLndInvoice.js (hunk 5 above)
Comment (Overtorment) [reply screen/lnd/scanLndInvoice.js]: ok, just tested, this stuff breaks going back if lnd invocie successfully paid. alert shows "success" and screen stays the same. revert all such cases.

**Comment by Overtorment** [comment]
@marcosrdz try not to mix different features in a single branch, its hard to review, hard to track whats going on, etc. or if you really like to mix things, we can try working with release branches, for example, create branch `release-3-5-0` and work on that till its good enough, then merge to master and cut a release

File: screen/wallets/list.js [hunk 8]
```diff
@@ -337,6 +413,11 @@ export default class WalletsList extends Component {
                         navigate('TransactionDetails', {
                           hash: rowData.item.hash,
                         });
+                      } else if (rowData.item.type === 'user_invoice') {
+                        // this.props.navigation.navigate('LNDViewInvoice', {
+                        //   invoice: rowData.item,
```
Comment (Overtorment) [inline screen/wallets/list.js]: TODO: correct wallet lookup

File: class/app-storage.js [hunk 9]
```diff
@@ -174,8 +174,10 @@ export class AppStorage {
               break;
           }
           // done
-          this.wallets.push(unserializedWallet);
-          this.tx_metadata = data.tx_metadata;
+          if (!this.wallets.some(wallet => wallet.getSecret() === unserializedWallet.secret)) {
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply class/app-storage.js]: yup

File: ios/BlueWallet.xcodeproj/xcshareddata/xcschemes/BlueWallet.xcscheme [hunk 10]
```diff
@@ -80,7 +80,7 @@
       </AdditionalOptions>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Release"
+      buildConfiguration = "Debug"
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply ios/BlueWallet.xcodeproj/xcshareddata/xcschemes/BlueWallet.xcscheme]: Xcode being Xcode.

File: ios/BlueWallet/Info.plist (hunk 1 above)
Comment (marcosrdz) [reply ios/BlueWallet/Info.plist]: Xcode being Xcode.

File: screen/lnd/lndCreateInvoice.js (hunk 7 above)
Comment (marcosrdz) [reply screen/lnd/lndCreateInvoice.js]: Fixed.

File: MainBottomTabs.js [hunk 11]
```diff
@@ -194,12 +249,19 @@ const MainBottomTabs = createStackNavigator(
...
+        header: null,
+      },
+    },
+    LNDViewExistingInvoice: {
+      screen: LNDViewInvoiceStackNavigator,
+      navigationOptions: {
+        header: null,
+      },
     },
+    // Select Wallet. Mostly for deep-linking
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply MainBottomTabs.js]: Fixed.

**Pull request merged**